### PR TITLE
Fix windows ransomware readme

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Fixes line endings in Ransomware README in Windows.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Pending manual tests by @cakekoa 
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by cakekoa — APPROVED** [review]

File: monkey/agent_plugins/payloads/ransomware/src/readme_dropper.py [hunk 1]
```diff
@@ -2,13 +2,35 @@
...
+        if dest.exists():
+            logger.warning(f"{dest} already exists, not leaving a new README.txt")
+            return
+
+        logger.info(f"Leaving a ransomware README file at {dest}")
+
+        if self._operating_system == OperatingSystem.WINDOWS:
+            self._leave_windows_readme(src, dest)
+        else:
+            self._leave_posix_readme(src, dest)
```
Comment (shreyamalviya) [inline monkey/agent_plugins/payloads/ransomware/src/readme_dropper.py:24]: Could just do
```python
class ReadmeDropper:
    def __init__(self, operating_system: OperatingSystem):
        if operating_system == OperatingSystem.WINDOWS:
            self._leave_readme = self._leave_windows_readme
        else:
            self._leave_readme = self._leave_posix_readme

    def leave_readme(self, src: Path, dest: Path):
        if dest.exists():
            logger.warning(f"{dest} already exists, not leaving a new README.txt")
            return

        logger.info(f"Leaving a ransomware README file at {dest}")
        
        self._leave_readme(src, dest)
```

Comment (mssalvatore) [reply monkey/agent_plugins/payloads/ransomware/src/readme_dropper.py:24]: As written, it's less cognitive load on the developer to be able to read `leave_readme()` and understand exactly what's happening.

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/tests/unit_tests/agent_plugins/payloads/ransomware/test_readme_dropper.py [hunk 2]
```diff
@@ -2,11 +2,14 @@
...
 import pytest
-from agent_plugins.payloads.ransomware.src.readme_dropper import leave_readme
+from agent_plugins.payloads.ransomware.src.readme_dropper import ReadmeDropper
 from tests.utils import get_file_sha256_hash
 
+from common import OperatingSystem
+
 DEST_FILE = "README.TXT"
 EMPTY_FILE_HASH = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+WINDOWS_README_FILE_HASH = "4231329e0212592178f9d590cdc0cfcf4dc691b135e2cb4fc49227b17d32ca8c"
```
Comment (lifeofzi) [inline monkey/tests/unit_tests/agent_plugins/payloads/ransomware/test_readme_dropper.py:12]: @mssalvatore What is the test evironment that passes this - I am not able to generate the same hash - 

Here is what I get 

```
<?xml version="1.0" encoding="utf-8"?><testsuites name="pytest tests"><testsuite name="pytest" errors="0" failures="1" skipped="0" tests="3" time="0.216" timestamp="2025-10-25T07:07:22.881660+00:00" hostname="16a28c27e892"><testcase classname="" name="test_readme_already_exists" time="0.009" /><testcase classname="" name="test_leave_readme_linux" time="0.002" /><testcase classname="" name="test_leave_readme_windows" time="0.005"><failure message="AssertionError: assert '92b772380a3f...b20de10a66925' == '4231329e0212...227b17d32ca8c'&#10;  &#10;  - 4231329e0212592178f9d590cdc0cfcf4dc691b135e2cb4fc49227b17d32ca8c&#10;  + 92b772380a3f8e27a93e57e6deeca6c01da07f5aadce78bb2fbb20de10a66925">src_readme = PosixPath('/dev/monkey/tests/data_for_tests/test_readme.txt')
dest_readme = PosixPath('/tmp/pytest-of-root/pytest-0/test_leave_readme_windows0/README.TXT')

    def test_leave_readme_windows(src_readme: Path, dest_readme: Path):
        readme_dropper = ReadmeDropper(OperatingSystem.WINDOWS)
    
        readme_dropper.leave_readme(src_readme, dest_readme)
    
&gt;       assert get_file_sha256_hash(dest_readme) == WINDOWS_README_FILE_HASH
E       AssertionError: assert '92b772380a3f...b20de10a66925' == '4231329e0212...227b17d32ca8c'
E         
E         - 4231329e0212592178f9d590cdc0cfcf4dc691b135e2cb4fc49227b17d32ca8c
E         + 92b772380a3f8e27a93e57e6deeca6c01da07f5aadce78bb2fbb20de10a66925

monkey/tests/unit_tests/agent_plugins/payloads/ransomware/test_readme_dropper.py:47: AssertionError</failure></testcase></testsuite></testsuites>
```

Comment (mssalvatore) [reply monkey/tests/unit_tests/agent_plugins/payloads/ransomware/test_readme_dropper.py:12]: @lifeofzi What branch are you on? This test has been removed on the `develop` branch.

Comment (lifeofzi) [reply monkey/tests/unit_tests/agent_plugins/payloads/ransomware/test_readme_dropper.py:12]: ah!
Thanks!

**Pull request merged**